### ISSUE TITLE
Missing quotes on joined table aliases

### ISSUE THREAD

**Issue by jeromenerf** [go-pg/pg#422]
After #393, I updated `go-pg` to the latest commit.

Given `User` (default table name `users`) and `ContractUser` (many to many relation table, default table name `contract_users`): 
```
    users := []User{}
    err = app.DB.Model(&[]ContractUser{}).
      Column("User").
      Where(`contract_id = ? AND "contract_user".role = 'WORKER'`, contract.ID).                                                                                                                
      Order("random()").
      Limit(1 + rand.Intn(5)).
      Select(&users)
```

produces this query:

```
# SELECT "contract_user"."id", "contract_user"."contract_id", "contract_user"."user_id", "contract_user"."role", "contract_user"."created_at", "contract_user"."updated_at", user."id" AS "user__id", user."user_name" AS "user__user_name", user."last_name" AS "user__last_name", user."first_name" AS "user__first_name", user."email" AS "user__email", user."is_admin" AS "user__is_admin", user."role" AS "user__role", user."geom" AS "user__geom", user."password" AS "user__password", user."created_at" AS "user__created_at", user."updated_at" AS "user__updated_at" FROM "contract_users" AS "contract_user" LEFT JOIN "users" AS user ON user."id" = "contract_user"."user_id" WHERE (contract_id = 1 AND "contract_user".role = 'WORKER') ORDER BY "random()" LIMIT 1;
```

which returns this error: 

```
ERROR:  syntax error at or near "."
LINE 1: ..."."created_at", "contract_user"."updated_at", user."id" AS "...
```

The `users` table is aliased to "user" but not quoted. Hence `user."id"` fails.

**Comment by jeromenerf** [comment]
NB1: using `"user"` fixes the problem. 

NB2: the `random()` function is also incorrectly quoted there.

**Comment by vmihailenco** [comment]
>The users table is aliased to "user" but not quoted. Hence user."id" fails.

I was not aware of this (and still did not have enough time to fully confirm it). I guess we should always quote table name/alias then?

>NB2: the random() function is also incorrectly quoted there.

You should use `OrderExpr("random()")` instead. `Order` quotes passed column name so you don't have to worry about SQL injections when you pass data from `request.URL` or some 3rd party sources. I tried to document this change in https://github.com/go-pg/pg/blob/v5/CHANGELOG.md#v5

**Comment by jeromenerf** [comment]
NB1: In this particular case, it would be correct. 

NB2: concerning, OrderExpr, I missed this point, my bad. This is logical and consistent.

Thanks!

**Comment by jeromenerf** [comment]
@vmihailenco Corner case: `user` being a reserved word, quoting it is mandatory. 

Double quoting seems safe as per https://www.postgresql.org/docs/9.5/static/sql-syntax-lexical.html

**Comment by jeromenerf** [comment]
As a workaround, I settled on renaming "User" to "Person" and go on with it. 

However, by importing `github.com/jinzhu/inflection`, `Person` gets pluralized to `People`, creating another level of brainfuck TT

**Comment by jeromenerf** [comment]
@vmihailenco Have you been able to reproduce this issue?

Just in case, here is a small snippet to reproduce it: 

### Context

The m2m relationship is not mandatory.

```
package main

import (
	"log"
	"os"

	pg "gopkg.in/pg.v5"
)

// User is a reserved sql keyword
type User struct {
	ID         int64
	Name       string
	Issues     []*Issue `pg:",many2many:user_issues"`
}

// UserIssue links user to issue with role
type UserIssue struct {
	ID      int64
	Role    string
	UserID  int64
	IssueID int64
	User    *User
	Issue   *Issue
}

// Issue has users
type Issue struct {
	ID         int64
	Title      string
	Users      []*User `pg:",many2many:user_issues"`
}

func main() {
	pg.SetQueryLogger(log.New(os.Stdout, "", log.LstdFlags))

	db := pg.Connect(&pg.Options{
		User:     "USER",
		Password: "PASSWORD",
	})

	queries := []string{
		`CREATE TEMP TABLE users (id serial, name text)`,
		`CREATE TEMP TABLE issues (id serial, title text)`,
		`CREATE TEMP TABLE user_issues (id serial, role text, user_id bigint, issue_id bigint)`,
	}
	for _, q := range queries {
		_, err := db.Exec(q)
		if err != nil {
			return
		}
	}

	users := []User{}
	err := db.
		Model(&[]UserIssue{}).
		Column("User").
		Select(&users)

	if err != nil {
		log.Println(err)
		return
	}

	log.Println(len(users))
}
```

### Expected

Return a list of users, in this case, all users linked to issues.


### Problem

The `User` model is linked to the `users` table, which is aliased to `user`. in the query, `user` is not quoted like `"user_issue"` and being a reserved word, il fails with: 

```
% go run foo.go
2016/11/29 11:53:41 foo.go:48: CREATE TEMP TABLE users (id serial, name text)
2016/11/29 11:53:41 foo.go:48: CREATE TEMP TABLE issues (id serial, title text)
2016/11/29 11:53:41 foo.go:48: CREATE TEMP TABLE user_issues (id serial, role text, user_id bigint, issue_id bigint)
2016/11/29 11:53:41 foo.go:58: SELECT "user_issue"."id", "user_issue"."role", "user_issue"."user_id", "user_issue"."issue_id", user."id" AS "user__id", user."name" AS "user__name" FROM "user_issues" AS "user_issue" LEFT JOIN "users" AS user ON user."id" = "user_issue"."user_id"
2016/11/29 11:53:41 ERROR #42601 syntax error at or near "." (addr="127.0.0.1:5432")
```

### NB

Older v5 code used to quote the aliases. Moreover, for some reason, the query was also more compact. 

I don't really see why the "user_issue" columns are present in the SELECT part there, since `.Column("User")` doen't include `user_issue.*`.

**Comment by vmihailenco** [comment]
I fixed alias quotation issue, but column names require more thinking and work.

**Comment by jeromenerf** [comment]
Agreed. I'll try and test it out later this week. 
Thanks!

**Comment by vmihailenco** [comment]
I fixed this by documenting current behaviour - https://github.com/go-pg/pg#column-names. Please open new issue if you think there are uncovered use cases.

**Comment by clwilliams** [comment]
I just experienced the error as @vmihailenco with an orders table, joining to some other tables using a Relation query. All the tables have an id field; when searching for a specific order by id, it wanted order.id to be specified to make it clearer which id I meant and I got the error above "syntax error at or near "."". Renaming the table worked, which is a bit overkill but trying to alias the table name or column didn't work. I knew it was specific to querying on id and the query I run to get all the orders fo an account works, but as soon as I add the where clause order.id, I had an error.
